SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp

```diff
@@ -44,27 +45,70 @@ EntityFactory::EntityFactory(USpatialNetDriver* InNetDriver, USpatialPackageMapC
 {
 }
 
-TArray<FWorkerComponentData> EntityFactory::CreateEntityComponents(USpatialActorChannel* Channel,
-																   FRPCsOnEntityCreationMap& OutgoingOnCreateEntityRPCs,
-																   uint32& OutBytesWritten)
+EntityComponents EntityFactory::CreateSkeletonEntityComponents(AActor* Actor)
 {
-	AActor* Actor = Channel->Actor;
 	UClass* Class = Actor->GetClass();
-	Worker_EntityId EntityId = Channel->GetEntityId();
 
-	FString ClientWorkerAttribute = GetConnectionOwningWorkerId(Actor);
+	WorkerRequirementSet ReadAcl;
+	if (Class->HasAnySpatialClassFlags(SPATIALCLASS_ServerOnly))
+	{
+		ReadAcl = SpatialConstants::UnrealServerPermission;
+	}
+	else
+	{
+		ReadAcl = SpatialConstants::ClientOrServerPermission;
+	}
 
-	WorkerRequirementSet AnyServerRequirementSet = { SpatialConstants::UnrealServerAttributeSet };
-	WorkerRequirementSet AnyServerOrClientRequirementSet = { SpatialConstants::UnrealServerAttributeSet,
-															 SpatialConstants::UnrealClientAttributeSet };
+	WriteAclMap ComponentWriteAcl;
+	ComponentWriteAcl.Add(SpatialConstants::ENTITY_ACL_COMPONENT_ID, SpatialConstants::UnrealServerPermission);
+
+	EntityComponents EntityComps;
+	EntityComps.ModifiableComponents.Add(SpatialConstants::POSITION_COMPONENT_ID,
+										 MakeUnique<Position>(Coordinates::FromFVector(GetActorSpatialPosition(Actor))));
+	EntityComps.ModifiableComponents.Add(SpatialConstants::METADATA_COMPONENT_ID, MakeUnique<Metadata>(Class->GetName()));
+	EntityComps.ModifiableComponents.Add(SpatialConstants::SPAWN_DATA_COMPONENT_ID, MakeUnique<SpawnData>(Actor));
+
+	if (ShouldActorHaveVisibleComponent(Actor))
+	{
+		EntityComps.ComponentDatas.Add(ComponentFactory::CreateEmptyComponentData(SpatialConstants::VISIBLE_COMPONENT_ID));
+	}
+
+	if (!Class->HasAnySpatialClassFlags(SPATIALCLASS_NotPersistent))
+	{
+		EntityComps.ComponentDatas.Add(Persistence().CreateComponentData());
+	}
+
+	if (Actor->NetDormancy >= DORM_DormantAll)
```

Comment:
this is confusing. Isn't dormancy an unreal component? As well as things like metadata and spawndata (but this function isn't the unreal components one)

Reply:
Hm, I tried to include all things we can infer just by having an Actor pointer.
Don't mind too much wherever it goes.

Reply:
is that list of things defined somewhere? What benefits do we get for including it here?

Reply:
i.e. does it simplify logic elsewhere (if it's otherwise unused and only here because it can be would prefer dropping it to keep this simple)